### PR TITLE
CI: Use g++ 4.8 in ubuntu-18.04 builds

### PR DESCRIPTION
**Pull request by ccawley2011**
This should help catch issues with older versions of GCC in pull requests.

### CODE REVIEW

**Comment by AndywinXp** [comment]
Yes, please 👀

**Comment by dwatteau** [comment]
Nice! But nore that the 18.04 image will be deprecated with brownout periods very soon:
https://github.com/actions/virtual-environments/issues/6002

**Comment by ccawley2011** [comment]
> Nice! But note that the 18.04 image will be deprecated with brownout periods very soon: [actions/virtual-environments#6002](https://github.com/actions/virtual-environments/issues/6002)

I've upgraded to Ubuntu 20.04, and worked around the lack of GCC 4.8 by manually adding the package repositories from Xenial.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -149,15 +149,22 @@ jobs:
...
+            aptPackages: 'g++-4.8 libsdl1.2-dev libsdl-net1.2-dev liba52-dev libjpeg-turbo8-dev libmpeg2-4-dev libogg-dev libvorbis-dev libflac-dev libmad0-dev libpng-dev libtheora-dev libfaad-dev libfluidsynth-dev libfreetype6-dev zlib1g-dev libfribidi-dev libcurl4-openssl-dev libgtk-3-dev libspeechd-dev libsndio-dev libunity-dev'
     env:
       SDL_CONFIG: ${{ matrix.sdlConfig }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+      - name: Add Ubuntu Xenial package sources
+        if: matrix.platform == 'ubuntu-20.04'
+        run: |
+          sudo apt-add-repository 'deb http://dk.archive.ubuntu.com/ubuntu/ xenial main'
```
Comment (lephilousophe) [inline .github/workflows/ci.yml:166]: Maybe remove the dk part

Comment (bluegr) [inline .github/workflows/ci.yml:166]:
```suggestion
          sudo apt-add-repository 'deb http://archive.ubuntu.com/ubuntu/ xenial main'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -149,15 +149,22 @@ jobs:
...
+          sudo apt-add-repository 'deb http://dk.archive.ubuntu.com/ubuntu/ xenial universe'
```
Comment (bluegr) [inline .github/workflows/ci.yml:167]:
```suggestion
          sudo apt-add-repository 'deb http://archive.ubuntu.com/ubuntu/ xenial universe'
```

**Comment by bluegr** [comment]
Thanks for your work! This will come in handy indeed. The leftover issues with the Ubuntu archive links. have been fixed.
Merging

**Pull request merged**